cachi2/core/package_managers/gomod.py

```diff
@@ -335,6 +340,24 @@ def go_list_deps(pattern: Literal["./...", "all"]) -> Iterator[GoPackage]:
     return main_module, all_modules, all_packages
 
 
+def _deduplicate_resolved_modules(modules: Iterable[GoModule]) -> list[GoModule]:
+    def get_unique_key(module: GoModule) -> tuple[str, str]:
+        if module.replace and not module._is_local_replacement:
+            return module.replace.path, module.replace.version
+
+        return (module.path, module.version)
```

Comment:
Why not use (the code from) `_get_name_and_version` as the unique key? Will this properly de-duplicate locally replaced modules?

Reply:
I used a slightly different version of it (and yet another slightly different version of it on the _create_modules_from_parsed_data function) :)